app/scripts/models/flow.js

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent a login flow, holding the data that we
+ * need to submit as part of our metrics when we perform actions
+ * during the flow.
+ *
+ * Try to fetch the flowId and flowBegin from the resume token.
+ * If unavailable there, fetch from data attributes in the DOM.
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var AuthErrors = require('lib/auth-errors');
+  var Backbone = require('backbone');
+  var Cocktail = require('cocktail');
+  var ErrorUtils = require('lib/error-utils');
+  var ResumeTokenMixin = require('models/mixins/resume-token');
+  var SearchParamMixin = require('models/mixins/search-param');
+  var vat = require('lib/vat');
+
+  var Model = Backbone.Model.extend({
+    initialize: function (options) {
+      options = options || {};
+
+      this.sentryMetrics = options.sentryMetrics;
+      this.window = options.window || window;
+
+      this.fetch();
+    },
+
+    defaults: {
+      flowBegin: null,
+      flowId: null
+    },
+
+    fetch: function () {
+
+      var error;
+      var $body = this.window['$']('body');
```

Comment:
Instead of using `this.window['$']`, at the top can you create a reference to jQuery, and use that here?

```js
var $ = require('jquery');


...

var $body = $(this.window).find('body');
```

I think that last find should work anyways.